classes/utils/Utils.py

```diff
@@ -0,0 +1,321 @@
+import re
+
+from enum import Enum, IntEnum
+
+
+class Version(Enum):
+
+    def __init__(self, major, minor, version, protocol):
+        self.major = major
+        self.minor = minor
+        self.version = version
+        self.protocol = protocol
+        self.all = (major, minor, version)
+
+    def is_same_major(self, other_version):
+        return self.major == other_version.major
+
+    def get_version(protocol):
+        for v in Version:
+            if v.protocol == protocol:
+                return v
+        return None
+
+    def __lt__(self, other):
+        if type(other) == tuple:
+            return self.all < other
+        else:
+            return self.all < other.all
+
+    def __le__(self, other):
+        return not self.__gt__(other)
+
+    def __eq__(self, other):
+        if type(other) == tuple:
+            return self.all == other
+        else:
+            return self.all == other.all
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __ge__(self, other):
+        return not self.__lt__(other)
+
+    def __gt__(self, other):
+        if type(other) == tuple:
+            return self.all > other
+        else:
+            return self.all > other.all
+
+    v1_12_0 = (1, 12, 0, 335)
+    v1_12_1 = (1, 12, 1, 338)
+    v1_12_2 = (1, 12, 2, 340)
+    v1_13_0 = (1, 13, 0, 393)
+    v1_13_1 = (1, 13, 1, 401)
+    v1_13_2 = (1, 13, 2, 404)
+    v1_14_0 = (1, 14, 0, 477)
+    v1_14_1 = (1, 14, 1, 480)
+    v1_14_2 = (1, 14, 2, 485)
+    v1_14_3 = (1, 14, 3, 490)
+    v1_14_4 = (1, 14, 4, 498)
+    v1_15_0 = (1, 15, 0, 573)
+    v1_15_1 = (1, 15, 1, 575)
+    v1_15_2 = (1, 15, 2, 578)
+    v1_16_0 = (1, 16, 0, 735)
+    v1_16_1 = (1, 16, 1, 736)
+    v1_16_2 = (1, 16, 2, 751)
+
+
+class GameMode(IntEnum):
+
+    SURVIVAL = 0
+    CREATIVE = 1
+    ADVENTURE = 2
+    SPECTATOR = 3
+
+
+regex_color = re.compile(r"§[0-9A-FK-OR]", re.IGNORECASE)
+
+
+class Color(Enum):
```

Comment:
Use chatcolor instead. There is a name clash with Color (Like display color - used internally) and a implicit one with Map Color (used for maps in minecraft).
I'd also notate that it's legacy and should not be used.